virtqemud_t fixes from cockpit tests (bsc#1242998)

ca-hu opened the pull request:
These AVCs were found during cockpit tests on SUSE, might be interesting for fedora as well

Use case:
- set up cockpit
- on cockpit web interface create and start a virtual machine
- see denials in audit log

Versions:
- libvirt 11.2.0
- cockpit 334
- policy: opensuse selinux-policy with fedora commits until 23514206ea45e1d1d2f8a4c08288065c813fcc91

ca-hu committed:
Allow virtqemud_t to read and write generic pty (bsc#1242998)

Adresses:
type=AVC msg=audit(1751541756.409:444): avc:  denied  { open } for  pid=3862 comm="qemu-system-x86" path="/dev/pts/1" dev="devpts" ino=4 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:object_r:devpts_t:s0 tclass=chr_file permissive=1

ca-hu committed:
Allow virtqemud_t setattr to /dev/userfaultfd (bsc#1242998)

https://github.com/libvirt/libvirt/blob/01badd1bfb131a180713dd89abec20acdcd333f0/NEWS.rst?plain=1#L1190

Adresses:
----
time->Thu Jul  3 13:22:36 2025
type=AVC msg=audit(1751541756.384:440): avc:  denied  { setattr } for  pid=3865 comm="rpc-virtqemud" name="userfaultfd" dev="tmpfs" ino=7 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:object_r:userfaultfd_device_t:s0 tclass=chr_file permissive=1

ca-hu committed:
Extend virtqemud_t tcp_socket permissions (bsc#1242998)

...to listen and accept

Adresses:
----
time->Thu Jul  3 13:22:36 2025
type=AVC msg=audit(1751541756.434:446): avc:  denied  { listen } for  pid=3862 comm="qemu-system-x86" laddr=127.0.0.1 lport=5900 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:system_r:virtqemud_t:s0 tclass=tcp_socket permissive=1
----
time->Thu Jul  3 13:22:36 2025
type=AVC msg=audit(1751541756.806:455): avc:  denied  { accept } for  pid=3862 comm="qemu-system-x86" laddr=127.0.0.1 lport=5900 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:system_r:virtqemud_t:s0 tclass=tcp_socket permissive=1

ca-hu committed:
Allow virtqemud_t read and write /dev/ptmx (bsc#1242998)

Adresses:

type=AVC msg=audit(1751541756.408:441): avc:  denied  { read write } for  pid=3862 comm="qemu-system-x86" name="ptmx" dev="tmpfs" ino=9 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:object_r:ptmx_t:s0 t
class=chr_file permissive=1
type=AVC msg=audit(1751541756.409:442): avc:  denied  { open } for  pid=3862 comm="qemu-system-x86" path="/dev/ptmx" dev="tmpfs" ino=9 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:object_r:ptmx_t:s0 tclass=chr_file permissive=1
type=AVC msg=audit(1751541756.409:443): avc:  denied  { ioctl } for  pid=3862 comm="qemu-system-x86" path="/dev/ptmx" dev="tmpfs" ino=9 ioctlcmd=0x5430 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:object_r:ptmx_t:s0 tclass=chr_file permissive=1

zpytela commented on policy/modules/contrib/virt.te:
```diff
@@ -2292,6 +2293,7 @@ fs_setattr_tmpfs_chr_blk_files(virtqemud_t)
 fs_relabel_tmpfs_blk_file(virtqemud_t)
 fs_relabel_tmpfs_chr_file(virtqemud_t)
 fs_relabelfrom_tmpfs_lnk_files(virtqemud_t)
+fs_rw_inherited_tmpfs_files(virtqemud_t)
```
Which service created the memfd:displaysurface file?

ca-hu replied:
qemu creates it as far as i could see: https://github.com/qemu/qemu/blob/e240f6cc25917f3138d9e95e0343ae23b63a3f8c/ui/console.c#L494

zpytela replied:
So it can use its private virtqemud_tmpfs_t type, right?

ca-hu replied:
yep changed it, thanks :)

zpytela commented on policy/modules/contrib/virt.te:
```diff
@@ -2307,6 +2309,9 @@ storage_manage_fixed_disk(virtqemud_t)
 
+term_use_generic_ptys(virtqemud_t)
+term_use_ptmx(virtqemud_t)
+
```
Is there a specific use case when these terms are needed? Is it worth having it behind a boolean?

ca-hu replied:
its default that the serial console is connected to /dev/pts/x, so i dont think it is a special use case

ca-hu committed:
Allow virtqemud_t use its private tmpfs files (bsc#1242998)

Adresses
----
time->Thu Jul  3 13:22:36 2025
type=AVC msg=audit(1751541756.425:445): avc:  denied  { write } for  pid=3862 comm="qemu-system-x86" name="memfd:displaysurface" dev="tmpfs" ino=255 scontext=system_u:system_r:virtqemud_t:s0 tcontext=system_u:object_r:tmpfs_t:s0 tclass=file permissive=1

zpytela commented:
I believe all issues have been resolved now.

ca-hu commented:
yes, thank you :)

zpytela commented:
Merging, thank you.
